code/update_json

```diff
@@ -4,55 +4,108 @@ import json
-            in_struct = json.load(f)
+def analyze_struct(bytes_struct):
+    # figure out original indentation
+    indent = 0
+    indent_sym = None
+    multiline = False
+    newline = None
 
-        # figure out original indentation
-        indent = 0
-        multiline = False
-        with open(infile, 'r', encoding='utf-8') as f:
-            for i, line in enumerate(f.readlines()):
-                if i:
-                    multiline = True
-                if not line.startswith('{'):
-                    for s in line:
-                        if s != ' ':
-                            break
-                        indent += 1
+    for i, line in enumerate(bytes_struct.splitlines(keepends=True)):
+        if i:
+            multiline = True
+        else:
+            for e in (b'\r\n', b'\n'):
+                if line.endswith(e):
+                    newline = e
+                    break
+        if newline:
+            line = line.rstrip(newline)
+
+        line = line.decode('utf-8')
+        if not line.startswith('{'):
+            for j, s in enumerate(line):
+                if s not in (' ', '\t'):
                     break
-            assert indent < 16, f"got indent {indent}"
+                if indent_sym is None:
+                    indent_sym = s
+                if s != indent_sym:
+                    raise ValueError(f"Found mixed indentation of tabs and spaces in the line {line!r}")
+                indent += 1
+            break
+    assert indent < 16, f"got indent {indent} using {indent_sym}"
+
+    indent_ = indent_sym * indent if indent_sym else None
+
+    return indent_, multiline, newline
+
+def do_update_json(struct, annotated):
+
+    if isinstance(annotated, dict):
+        ann_struct = annotated
+    else:
+        with open(annotated, 'r', encoding='utf-8') as f:
+            ann_struct = json.load(f)
+
+    if struct.startswith('{'):
+        in_file = None
+        in_struct = json.loads(struct)
+        struct_bytes = struct.encode('utf-8')
+    elif op.exists(struct):
+        in_file = struct
+        with open(struct, 'rb') as f:
+            struct_bytes = f.read()
+        in_struct = json.loads(struct_bytes.decode('utf-8'))
 
+    if in_struct is not None:
         # we care only about Annotations
+        indent, multiline, newline = analyze_struct(struct_bytes)
+
         for k, v in in_struct.items():
             if target in v:
-                raise ValueError(f"File {infile} already has {target} value: {v[target]}")
+                raise ValueError(f"Input already has {target} value: {v[target]}")
             if target in ann_struct[k]:
                 v[target] = ann_struct[k][target]
 
         for k in ann_struct:
             if k not in in_struct:
                 in_struct[k] = ann_struct[k]
-
     else:
-        print(f"No {infile} -- we will produce one from scratch")
+        # if no input -- sensible defaults
+        indent = 4  # spaces
+        multiline = False
```

Comment:
```suggestion
        multiline = True
```
I think `True` makes more sense here - makes no difference either way but you always end up with a "multiline" file from this point on, so it's a bit less confusing